packages/app/components/checkout/select-plan.tsx

```diff
@@ -0,0 +1,155 @@
+import * as React from "react";
+import { useState, useEffect } from "react";
+
+import useSWRMutation from "swr/mutation";
+
+import { Button } from "@showtime-xyz/universal.button";
+import { ErrorText } from "@showtime-xyz/universal.fieldset";
+import { useIsDarkMode } from "@showtime-xyz/universal.hooks";
+import { Pressable } from "@showtime-xyz/universal.pressable";
+import { Skeleton } from "@showtime-xyz/universal.skeleton";
+import { Text } from "@showtime-xyz/universal.text";
+import { View } from "@showtime-xyz/universal.view";
+
+import { DropPlan, usePaidDropPlans } from "app/hooks/use-paid-drop-plans";
+import { axios } from "app/lib/axios";
+import { Logger } from "app/lib/logger";
+import { MY_INFO_ENDPOINT } from "app/providers/user-provider";
+
+export const SelectPlan = ({ setClientSecret }: { setClientSecret: any }) => {
+  const paidDropPlansQuery = usePaidDropPlans();
+  const [selectedPlan, setSelectedPlan] = useState<DropPlan | null>(null);
+  const isDark = useIsDarkMode();
+  const { trigger, data, isMutating, error } = useSWRMutation<{
```

Comment:
cc - @hirbod replaced it with a hook. Looks much better 😄

Reply:
We just need to keep in mind how the default error handling might kick in set in SWRConfig